src/lib/support/StringBuilder.h

```diff
@@ -0,0 +1,86 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <nlassert.h>
+
+#include "BufferWriter.h"
+
+namespace chip {
+
+/// Build a c-style string out of distinct parts
+class StringBuilderBase
+{
+public:
+    StringBuilderBase(char * buffer, size_t size) : mWriter(reinterpret_cast<uint8_t *>(buffer), size - 1)
+    {
+        nlASSERT(size > 0);
+        buffer[0] = 0; // make c-str work by default
+    }
+
+    /// Append a null terminated string
+    StringBuilderBase & Add(const char * s)
+    {
+        mWriter.Put(s);
+        NullTerminate();
+        return *this;
+    }
+
+    /// Append an integer value
+    StringBuilderBase & Add(int value)
+    {
+        char buff[32];
+        snprintf(buff, sizeof(buff), "%d", value);
+        buff[sizeof(buff) - 1] = 0;
```

Comment:
this one isn't necessary

also the largest `int value` should consume less than 32 bytes :-)

Reply:
Right ... even 64-bit is 20 chars at most. Wanted to not think about it since I imagine this code will be copied and pasted as if we ever want to format additional value types in here.

Do you think it would be useful to  drop the line or are ok to keep as is?